common/build/eslint-plugin-fluid/PUBLISH_GUIDE.md

```diff
@@ -0,0 +1,86 @@
+# Publishing Guide for `@fluidframework/eslint-config-fluid` & `@fluid-internal/eslint-plugin-fluid`
+
+This guide outlines the steps required to add new ESLint rules, update dependencies, and publish new versions of `@fluidframework/eslint-config-fluid` and `@fluid-internal/eslint-plugin-fluid`.
+
+## Steps
+
+### 1. Add New Rule to `eslint-plugin-fluid` (PR 1)
+
+1. **Create the Rule**: Write the new rule following best practices for ESLint. If you need guidance, refer to [ESLint's rule documentation](https://eslint.org/docs/latest/developer-guide/working-with-rules).
+	- [List of Custom Rules](https://github.com/microsoft/FluidFramework/tree/main/common/build/eslint-plugin-fluid/src/rules)
+
+2. **Testing**: Ensure the rule is thoroughly tested. Tests help validate that the rule behaves as expected across various code scenarios.
+
+   Directory structure:
+
+   ```plaintext
+   eslint-plugin-fluid/
+   ├── src
+   │   └── rules/					<!-- Contains the ESLint rule implementations -->
+   │       ├── rule-one.js
+   │       └── rule-two.js
+   ├── test/
+   │   ├── example/ 				<!-- Example mock files to test each rule in isolation -->
+   │   │   ├── rule-one/
+   │   │   │   ├── mockFileOne.js
+   │   │   │   └── mockFileTwo.js
+   │   │   └── rule-two/
+   │   │       ├── mockFileOne.js
+   │   │       └── mockFileTwo.js
+   │   ├── rule-one/				<!-- Test suite for rule-one -->
+   │   │   └── rule-one.test.js
+   │   └── rule-two/				<!-- Test suite for rule-two -->
+   │       └── rule-two.test.js
+
+3. **Update Changelog**: Record the new rule in the `CHANGELOG.md` file of the `@fluid-internal/eslint-plugin-fluid` package. This provides visibility into what was added for future reference.
+
+4. **Version Bump**: Update the package version following [Fluid Semantic Versioning](https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/dev/wiki/sync/breaking-vs-non-breaking-changes) guidelines. Typically:
+   - **Patch** version for fixes
+   - **Minor** version for new rules (backward-compatible)
+   - **Major** version for breaking changes
+
+### 2. Publish New Version of `eslint-plugin-fluid`
+
+Once PR 1 is merged, publish the new version of `@fluid-internal/eslint-plugin-fluid` by following the steps:
+
+1. **Publish**: Follow the [guide](https://eng.ms/docs/experiences-devices/opg/office-shared/fluid-framework/fluid-framework-internal/fluid-framework/docs/on-call/release/release) to publish the new version of `@fluid-internal/eslint-plugin-fluid`.
```

Comment:
Let's just point to "the guidance in our internal documentation" but avoid the link. We probably shouldn't link directly to internal documentation from the public repo.

Reply:
Probably note that publishing needs to be done by a Microsoft Fluid team member too, just in case an open-source developer reads this.

Reply:
@alexvy86 

Also removed the internal link for `Fluid Semantic Versioning`.